pkgs/top-level/static.nix

```diff
@@ -82,14 +82,15 @@ in {
   } // optionalAttrs super.stdenv.hostPlatform.isDarwin {
     pythonSupport = false;
   });
-  zlib = (super.zlib.override {
+  zlib = super.zlib.override {
     static = true;
     shared = false;
+    splitStaticOutput = false;
 
     # Don’t use new stdenv zlib because
     # it doesn’t like the --disable-shared flag
     stdenv = super.stdenv;
-  }).static;
```

Comment:
I just learned that this change accidentally also fixed more severe issues (https://github.com/nh2/static-haskell-nix/issues/73):

`nix-shell -p 'pkgsStatic.zlib.override {}' ` did not work until this commit, it errored:

```
error: attribute 'override' missing, at (string):1:94
```

FYI @malbarbo @dtzWill @matthewbauer

It appears that we must never use `.static` or other split outputs in things like `static.nix` because they are not overridable. (copied from https://github.com/nh2/static-haskell-nix/issues/73#issuecomment-580490912)

---

Also interesting is that `pkgsStatic.zlib` has nonsensical attributes like `cmakeFlags` (even though it does not use cmake) from here in the `make-derivation.nix` case of `lib.optionalAttrs (stdenv.hostPlatform != stdenv.buildPlatform)`: https://github.com/nh2/nixpkgs/blob/11aa987ea5b5a593c9ca7a38b391804959f905e5/pkgs/stdenv/generic/make-derivation.nix#L245-L254